### PR TITLE
Mes 5845 enforce serious dangerous faults code 4 and 5

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
## Description

Implements the logic from #1826 but for the remaining categories. CPC is exempt as per the user story https://jira.dvsacloud.uk/browse/MES-5845

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

File: src/pages/non-pass-finalisation/cat-a-mod2/__tests__/non-pass-finalisation.cat-a.mod2.page.spec.ts [hunk 1]
```diff
@@ -120,13 +134,71 @@ describe('NonPassFinalisationCatAMod2Page', () => {
...
+        component.continue();
+
+        // Assert
+        expect(component.openTestDataValidationModal).toHaveBeenCalled();
+        expect(component.modalController.create).toHaveBeenCalled();
+        expect(store$.dispatch).not.toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
+      });
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 4 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-a-mod2/__tests__/non-pass-finalisation.cat-a.mod2.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-a-mod2/__tests__/non-pass-finalisation.cat-a.mod2.page.spec.ts [hunk 2]
```diff
@@ -120,13 +134,71 @@ describe('NonPassFinalisationCatAMod2Page', () => {
         expect(store$.dispatch).toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
       });
 
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 5 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-a-mod2/__tests__/non-pass-finalisation.cat-a.mod2.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-adi-part2/__tests__/non-pass-finalisation.cat-adi-part2.page.spec.ts [hunk 3]
```diff
@@ -131,12 +154,69 @@ describe('NonPassFinalisationCatADIPart2Page', () => {
         expect(store$.dispatch).toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
       });
 
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 5 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-adi-part2/__tests__/non-pass-finalisation.cat-adi-part2.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-adi-part2/__tests__/non-pass-finalisation.cat-adi-part2.page.spec.ts [hunk 4]
```diff
@@ -131,12 +154,69 @@ describe('NonPassFinalisationCatADIPart2Page', () => {
...
+        component.continue();
+
+        // Assert
+        expect(component.openTestDataValidationModal).toHaveBeenCalled();
+        expect(component.modalController.create).toHaveBeenCalled();
+        expect(store$.dispatch).not.toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
+      });
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 4 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-adi-part2/__tests__/non-pass-finalisation.cat-adi-part2.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts [hunk 5]
```diff
@@ -123,12 +137,69 @@ describe('NonPassFinalisationCatBEPage', () => {
         expect(store$.dispatch).toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
       });
 
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 5 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts [hunk 6]
```diff
@@ -123,12 +137,69 @@ describe('NonPassFinalisationCatBEPage', () => {
...
+        component.continue();
+
+        // Assert
+        expect(component.openTestDataValidationModal).toHaveBeenCalled();
+        expect(component.modalController.create).toHaveBeenCalled();
+        expect(store$.dispatch).not.toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
+      });
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 4 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-c/__tests__/non-pass-finalisation.cat-c.page.spec.ts [hunk 7]
```diff
@@ -133,12 +147,69 @@ describe('NonPassFinalisationCatCPage', () => {
         expect(store$.dispatch).toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
       });
 
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 5 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-c/__tests__/non-pass-finalisation.cat-c.page.spec.ts]: Catb?

File: src/pages/non-pass-finalisation/cat-c/__tests__/non-pass-finalisation.cat-c.page.spec.ts [hunk 8]
```diff
@@ -133,12 +147,69 @@ describe('NonPassFinalisationCatCPage', () => {
...
+        component.continue();
+
+        // Assert
+        expect(component.openTestDataValidationModal).toHaveBeenCalled();
+        expect(component.modalController.create).toHaveBeenCalled();
+        expect(store$.dispatch).not.toHaveBeenCalledWith(new SetTestStatusWriteUp('123'));
+      });
+      it('should create the TestFinalisationInvalidTestDataModal when activityCode is 4 and no S/D faults', () => {
+        // Arrange
+        store$.dispatch(new testActions.StartTest(123, TestCategory.B));
```
Comment (andrewsetterfield) [inline src/pages/non-pass-finalisation/cat-c/__tests__/non-pass-finalisation.cat-c.page.spec.ts]: Catb?

**Review by andrewsetterfield — APPROVED** [review]

**Review by tomcrawleyy — APPROVED** [review]

**Pull request merged**